### PR TITLE
chmod, ls, uucore: Make chmod display human-friendly permissions

### PR DESCRIPTION
**Pull request by ProgVal**
when --changes or --verbose is given, like GNU chmod. Using on `ls`' code, which I moved to uucore.

### CODE REVIEW

**Comment by Arcterus** [comment]
Travis seems to still be failing on macOS.

**Comment by Arcterus** [comment]
Assuming the build succeeds on Windows, can you squash those commits?

**Comment by ProgVal** [comment]
I squashed the macOS fixes. Do you want me to squash the other commits as well?

And I didn't test on Windows (I don't have one to test on).

File: src/uucore/fs.rs [hunk 1]
```diff
@@ -195,3 +206,62 @@ pub fn is_stderr_interactive() -> bool {
...
+    String::from("---------")
+}
+
+#[cfg(unix)]
+pub fn display_permissions(metadata: &fs::Metadata) -> String {
+    let mode: mode_t = metadata.mode() as mode_t;
+    display_permissions_unix(mode as u32)
+}
+
+pub fn display_permissions_unix(mode: u32) -> String {
```
Comment (Arcterus) [inline src/uucore/fs.rs]: You need to add `#[cfg(unix)]` to this.

File: src/chmod/chmod.rs [hunk 2]
```diff
@@ -229,7 +230,7 @@ impl Chmoder {
     fn change_file(&self, fperm: u32, mode: u32, file: &Path, path: &str) -> Result<(), i32> {
```
Comment (Arcterus) [inline src/chmod/chmod.rs]: I think you also need to mark this as `#[cfg(unix)]` because you use `display_permissions_unix()`.

**Comment by Arcterus** [comment]
It's fine to just squash the ones you did.  I think this should be good to go when the Appveyor CI works again.

**Comment by Arcterus** [comment]
Thanks :)

**Pull request merged**